lib/Controller/Widget.php

```diff
@@ -241,6 +248,27 @@ public function addWidget(Request $request, Response $response, $type, $id)
         $displayOrder = $params->getInt('displayOrder');
         $playlist->assignWidget($widget, $displayOrder);
 
+        // Get the widget region and Layout
+        $widgetRegion = $this->regionFactory->getByPlaylistId($widget->playlistId)[0];
```

Comment:
Same as the previous comments - this can be a lot simpler.

```php
$playlist->load();
if ($playlist->isRegionPlaylist() && count($playlist->widgets) >= 1) {
    // Convert this region to a `playlist` (if it is a zone)
    $widgetRegion = $this->regionFactory->getById($playlist->regionId);
    (...)
}

```